src/shapes/paint/shape_paint.cpp

```diff
@@ -19,7 +18,10 @@ StatusCode ShapePaint::onAddedClean(CoreContext* context) {
 RenderPaint* ShapePaint::initRenderPaint(ShapePaintMutator* mutator) {
     assert(m_RenderPaint == nullptr);
     m_PaintMutator = mutator;
-    return m_RenderPaint = makeRenderPaint();
+
+    auto factory = mutator->component()->artboard()->factory();
```

Comment:
This gets hairy, but I get it. This is why part of me wanted to add an interface to inherit from if it needed a factory. Like 
```
class ShapePaint : public ShapePaintBase, public Fabricator {
   void provideFactory(Factory* factory) {
      // ok now make your render paint... and clean up old ones, allowing for swapping factories.
   }
};
```
Something at a higher level would need to know which classes are Fabricators (or whatever we call them) which Core could handle for us.

The only reason for this is to be able to swap factories at runtime or re-init with another factory (say you swap renderer or the rendering context is lost). But these could be dealt with in a different way too, this is just a pattern I've seen game engines use (initRenderer, disposeRenderer on any Graphics object).

Reply:
Your solution is nicer in that you're somewhat guaranteed things are available and you don't need to worry about async issues created by having a method that could theoretically be called at any time.